server/text_generation_server/utils/dist.py

```diff
@@ -57,7 +57,14 @@ def initialize_torch_distributed():
         options.is_high_priority_stream = True
         options._timeout = timedelta(seconds=60)
     else:
-        backend = "gloo"
+        try:
+            import oneccl_bindings_for_pytorch
+
+            backend = "ccl"
+            if os.getenv("CCL_WORKER_COUNT", None) is None:
```

Comment:
How does this env var need to be setup by the launcher?

Reply:
CCL_WORKER_COUNT=2 text-generation-launcher --model-id=rinna/japanese-gpt-neox-3.6b --num-shard 2 --sharded true. this could work. or we could put in in main.rs. what's your suggestion?

Reply:
What is CCL_WORKER_COUNT used for ?
Your reply seems to hint it's necessary for TP>1 ? Is that True ?

Reply:
yes, it's necessary for TP>1, Intel XPU use CCL to communicate within ranks, CCL_WORKER_COUNT is a must env for CCL